File: Samples/WVJavaSample/app/src/main/java/jp/supership/wvjavasample/WebAppInterface.java
package jp.supership.wvjavasample;

import android.content.Context;
import android.util.Log;
import android.webkit.JavascriptInterface;

import com.google.android.gms.ads.identifier.AdvertisingIdClient;

public class WebAppInterface {

    private static final String TAG = "WebAppInterface";
    private String advertising_id = "";
    private boolean isLAT = false;
    private Context context;

    public WebAppInterface(Context ct) {
        this.context = ct;
        this.initAdIdThread();
    }

    private void initAdIdThread() {
        Thread adIdThread =
                new Thread(
                        new Runnable() {
                            @Override
                            public void run() {
                                AdvertisingIdClient.Info adInfo = null;
                                try {
                                    adInfo = AdvertisingIdClient.getAdvertisingIdInfo(context);
                                    advertising_id = adInfo.getId();
                                    isLAT = adInfo.isLimitAdTrackingEnabled();
                                } catch (Exception e) {
                                    Log.d(TAG, "getAdId failed");
                                }
                            }
                        });
        adIdThread.start();
    }

    @JavascriptInterface
    public String getAppBundle() {
        return context.getPackageName();
    }

    @JavascriptInterface
    public String getAdvertisingId() {
        String adid = advertising_id;
        return !isLAT ? adid : "";
    }
}

File: Samples/WVKotlinSample/app/src/main/java/jp/supership/wvkotlinsample/WebAppInterface.kt
package jp.supership.wvkotlinsample

import android.content.Context
import android.os.Build
import android.util.Log
import android.webkit.JavascriptInterface
import com.google.android.gms.ads.identifier.AdvertisingIdClient

class WebAppInterface {

    companion object {
        private const val TAG = "WebAppInterface"
    }

    private var advertising_id: String? = ""
    private var isLAT: Boolean = false
    private lateinit var context: Context

    constructor(ct: Context) {
        this.context = ct
        this.initAdIdThread()
    }

    private fun initAdIdThread() {
        val adIdThread = Thread {
            var adInfo: AdvertisingIdClient.Info? = null
            try {
                adInfo = AdvertisingIdClient
                    .getAdvertisingIdInfo(this.context)
                advertising_id = adInfo.id
                isLAT = adInfo.isLimitAdTrackingEnabled
            } catch (e: Exception) {
                Log.d(TAG, "getAdId failed")
            }
        }
        adIdThread.start()
    }

    @JavascriptInterface
    open fun getAppBundle(): String {
        return this.context.packageName ?: ""
    }

    @JavascriptInterface
    open fun getAdvertisingId(): String {
        val adid = advertising_id ?: ""
        return if (!isLAT) adid else ""
    }

    @JavascriptInterface
    open fun getPlatformV(): String {
        return Build.VERSION.RELEASE
    }
}
